functions/src/roles/permissions.ts

```diff
@@ -0,0 +1,54 @@
+import { firestore } from "../admin/admin";
+import admin from "firebase-admin";
+import { Response, Request } from "express";
+
+interface permissionData {
+  documentName: string;
+  permission: string;
+}
+
+/**
+ * Add a single new permission to a role
+ * @param request
+ * @param response
+ */
+const addPermission = async (request: Request, response: Response): Promise<void> => {
```

Comment:
Might be easier just to use the update permission in most cases that way the frontend can be directed to one set route for managing existing permissions.